array.go

```diff
@@ -1193,3 +1193,143 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// IsOrdered succeeds if array is ordered based on optional `comparator` function.
+// For default, it will use built-in comparator function for each data type.
+// It requires all elements in the array to have same data type
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{100, 101, 102})
+//	array.IsOrdered(func(x, y *Value) bool {
+//		valX, _ := x.Raw().(int)
+//		valY, _ := y.Raw().(int)
+//		return valX >= valY
+//	})
+func (a *Array) IsOrdered(less ...func(x, y *Value) bool) *Array {
+	a.chain.enter("IsOrdered()")
+	defer a.chain.leave()
+
+	if a.chain.failed() {
+		return a
+	}
+
+	if len(less) > 1 {
+		a.chain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected multiple less arguments"),
+			},
+		})
+		return a
+	}
+
+	if len(a.value) == 0 {
+		return a
+	}
+
+	var fn func(x, y *Value) bool
+	if len(less) == 1 {
+		fn = less[0]
+	} else {
+		// use default comparator based on data type
+		cArr, ok := canonArray(a.chain, a.value)
+		if !ok {
+			return a
+		}
+
+		fn = a.getDefaultComparator(cArr)
+		if a.chain.failed() {
+			return a
+		}
+	}
+
+	for i := 0; i < int(a.Length().Raw())-1; i++ {
+		xChain := a.chain.clone()
+		xChain.replace("IsOrdered[%d]", i)
+		x := newValue(xChain, a.value[i])
+
+		yChain := a.chain.clone()
+		yChain.replace("IsOrdered[%d]", i+1)
+		y := newValue(yChain, a.value[i+1])
+
+		if !fn(x, y) {
```

Comment:
Since the function argument is called `less`, which is good because it's similar to sort functions from stdlib, let's treat it as `<`, not `<=`.

Hence, instead of `fn(x, y)` we should use `!fn(y, x)`.

i.e., `if !fn(x, y) {...}` should be replaced with `if fn(y, x) {...}`.

Reply:
Also, in default comparator we should use `<` instead of `<=` in all branches.